python/rikai/spark/functions/vision.py

```diff
@@ -117,7 +119,7 @@ def video_to_images(
     base_path = video.uri
 
     if isinstance(video, YouTubeVideo):
-        base_path = video.vid
+        base_path = os.path.join(tempfile.gettempdir(), video.vid)
```

Comment:
Thanks for addressing #130 .  This as in production code, will it make all data writting to the spark worker's local file system?

Reply:
I think what we want is an extra parameter in the UDF for the root. In the unit test you can just specifically pass in the path you get from the fixture.

Since this is actually materializing the images, I'm not sure whether you'll want to make this a local temporary directory. Otherwise any kind of action that causes shuffling/repartitioning will break this dataset.

1. Short term: add uri parameter to udf to address #130 
2. Separate from this PR, we should have a top-level rikai option that specifies a default temp s3 root. In production code the default temporary directory should come from this option
3. In the long-run, @eddyxu what do you think about having something like an "InMemoryImage" or allow `Image` to just have the in memory ndarray?

Reply:
Added a uri parameter for the YouTubeVideo case which defaults to tmp.

Reply:
@changhiskhan @smellslikeml do you foresee that whether each dataset (the output of one DataFrame) contains one of these asset directory, or there will be a global asset directory for all jobs? 

For example, could it be something like this:

```python

rikai.option.set("rikai.default.asset.directory", "s3://bucket/to/asset")

df.withColumn("image", video_to_image(df.video))  
```

Reply:
> In the long-run, @eddyxu what do you think about having something like an "InMemoryImage" or allow Image to just have the in memory ndarray?

yea, i think we can make `Image` compatible  to this, basically sth like

```
class Image(UDT):
    def __init__(self):
        self.content = None
        self.uri = None
        # we just keep one of them in the payload.
```